### PR TITLE
Jacoco plugin integration for unit tests

### PR DESCRIPTION
**Pull request by tharindu-bandara**
## Purpose
Generate unit test coverage information 

## Goals
Integrate Jacoco Maven plugin with Maven Surefire plugin to generate unit test coverage information

## Approach
N/A

## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
N/A

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
N/A

## Learning
N/A

### CODE REVIEW

File: components/data-services/org.wso2.carbon.dataservices.core/pom.xml [hunk 1]
```diff
@@ -1,5 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
+<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (chanikag) [inline components/data-services/org.wso2.carbon.dataservices.core/pom.xml]: Why the new line is removed? Is there any particular reason?

File: components/data-services/org.wso2.carbon.dataservices.core/pom.xml [hunk 2]
```diff
@@ -13,24 +12,20 @@
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
--->
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-
+--><project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (chanikag) [inline components/data-services/org.wso2.carbon.dataservices.core/pom.xml]: Same goes here..

**Comment by chanikag** [comment]
Can you do the modification only related to Integrating Jacoco Maven plugin, without doing any code formatting. It is hard to isolate what are the new changes by going through the PR. You may send a separate PR for the formatting changes.

File: components/data-services/org.wso2.carbon.dataservices.core/pom.xml (hunk 1 above)
Comment (tharindu-bandara) [reply components/data-services/org.wso2.carbon.dataservices.core/pom.xml]: Jacoco plugin was integrated using an automation tool and that tool perform code formatting

**Comment by tharindu-bandara** [comment]
Closed to create an another pull request with proper code formatting

**Pull request closed without merging**